python/ql/src/experimental/Security/CWE-022bis/TarSlipImprov.ql

```diff
@@ -100,19 +100,15 @@ class Configuration extends TaintTracking::Configuration {
   }
 
   override predicate isAdditionalTaintStep(DataFlow::Node nodeFrom, DataFlow::Node nodeTo) {
-    exists(AttrRead attr, MethodCallNode call |
-      attr.accesses(nodeFrom, "getmembers") and
-      nodeFrom = call.getObject() and
-      nodeFrom instanceof AllTarfileOpens and
-      nodeTo = call
-    )
+    nodeTo.(MethodCallNode).calls(nodeFrom, "getmembers") and
+    nodeFrom instanceof AllTarfileOpens
```

Comment:
I must resist the urge to suggest code-golfing this into 

```ql
nodeTo.(MethodCallNode).calls(nodeFrom.(AllTarfileOpens), "getmembers")
```

Reply:
Hah, that is shorter indeed, but I fear that clarity is being sacrificed too much..